Lesson4/src/com/home/PersonRegistry.java

```diff
@@ -0,0 +1,73 @@
+package com.home;
+
+import com.home.model.Address;
+import com.home.model.Person;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class PersonRegistry {
+    private Person[] citizens;
+
+    public PersonRegistry(Person[] citizens) {
+        this.citizens = citizens;
+    }
+
+    public PersonRegistry() {
+
+    }
+
+    public int countPeople(Address address) {
+        int count = 0;
+        for (Person person : citizens) {
+            Address personsAddress = person.getAddress();
+            if (personsAddress.getCountry().equals(address.getCountry())
+                    && personsAddress.getCity().equals(address.getCity())) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    public int countPeople(String country) {
+        int count = 0;
+        for (Person person : citizens) {
+            Address personsAddress = person.getAddress();
+            if (personsAddress.getCountry().equals(country)) {
+                count++;
+            }
+        }
+        return count;
+    }
+
+    public List<Person> getPeople(Address address) {
+        List<Person> people = new LinkedList<>();
+        for (Person person : citizens) {
+            Address personsAddress = person.getAddress();
+            if (personsAddress.getCountry().equals(address.getCountry())
+                    && personsAddress.getCity().equals(address.getCity())) {
+                people.add(person);
+            }
+        }
+        return people;
+    }
+
+    /**
+     * Show good recruits to army (not use)
+     */
+    public List<Person> getHealthyPeople() {
```

Comment:
this method should belong to the RecruitOffice and look a bit different.

In the RecruitOffice you want to get list of all people by address using method getPeople, and then do a sex and age check. 

By doing this you will not grant responsibility of RecruitOffice to the PersonRegistry, because it's not really their business to figure out who is fit for the military service